pkg/client/unversioned/helper.go

```diff
@@ -290,7 +291,8 @@ func SetKubernetesDefaults(config *Config) error {
 	if len(config.UserAgent) == 0 {
 		config.UserAgent = DefaultKubernetesUserAgent()
 	}
-	if len(config.Version) == 0 {
+	if len(config.Version) == 0 || apiutil.GetGroup(config.Version) != "" {
```

Comment:
@caesarxuchao per our discussion RESTClient and the ExperimentalClient don't share the same config. Therefore, this part should work as described above.